core/src/main/scala/org/apache/spark/scheduler/Stage.scala

```diff
@@ -91,3 +121,8 @@ private[spark] abstract class Stage(
     case _ => false
   }
 }
+
+private[scheduler] object Stage {
+  // The maximum number of times to retry a stage before aborting
+  private[scheduler] val MAX_STAGE_FAILURES = 4
```

Comment:
super minor, but you don't need the extra `private[scheduler]` when the whole object is already `private[scheduler]`